Realm/RLMSyncSessionRefreshHandle.mm

```diff
@@ -53,20 +53,38 @@ - (instancetype)initWithFullURLPath:(NSString *)urlPath
 }
 
 - (void)invalidate {
-    [self.timer invalidate];
-    self.user = nil;
+    dispatch_async(dispatch_get_main_queue(), ^{
```

Comment:
It's not necessary to `dispatch_async` here in order to call `-invalidate`. `NSTimer` is one of the classes listed as being thread safe.